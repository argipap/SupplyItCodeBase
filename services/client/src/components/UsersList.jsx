import React from 'react';

const UsersList = (props) => {
    return (
        <div>
            <h1 className="title is-1">All Users</h1>
            <hr/><br/>
            <table className="table is-hoverable is-fullwidth">
                <thead>
                <tr>
                    <th>ID</th>
                    <th>Email</th>
                    <th>Username</th>
                    <th>Active</th>
                    <th>Admin</th>
                </tr>
                </thead>
                <tbody>
                {
                    props.users.map((user) => {
                        return (
                            <tr key={user.id}>
                                <td>{user.id}</td>
                                <td>{user.email}</td>
                                <td>{user.username}</td>
                                <td>{String(user.active)}</td>
                                <td>{String(user.admin)}</td>
                            </tr>
                        )
                    })
                }
                </tbody>
            </table>
        </div>
    )
};

export default UsersList;