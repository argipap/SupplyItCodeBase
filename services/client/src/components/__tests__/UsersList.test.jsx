import React from 'react';
import {shallow} from 'enzyme';

import UsersList from '../UsersList';
import renderer from 'react-test-renderer';

const users = [
    {
        'active': true,
        'admin': false,
        'email': 'hermanmu@gmail.com',
        'id': 1,
        'username': 'michael'
    },
    {
        'active': true,
        'admin': false,
        'email': 'michael@mherman.org',
        'id': 2,
        'username': 'michaelherman'
    }
];

test('UsersList renders properly', () => {
    const wrapper = shallow(<UsersList users={users}/>);
    expect(wrapper.find('h1').get(0).props.children).toBe('All Users');
    // table
    const table = wrapper.find('table');
    expect(table.length).toBe(1);
    // table head
    expect(wrapper.find('thead').length).toBe(1);
    const th = wrapper.find('th');
    expect(th.length).toBe(5);
    expect(th.get(0).props.children).toBe('ID');
    expect(th.get(1).props.children).toBe('Email');
    expect(th.get(2).props.children).toBe('Username');
    expect(th.get(3).props.children).toBe('Active');
    expect(th.get(4).props.children).toBe('Admin');
    // table body
    expect(wrapper.find('tbody').length).toBe(1);
    expect(wrapper.find('tbody > tr').length).toBe(2);
    const td = wrapper.find('tbody > tr > td');
    expect(td.length).toBe(10);
    expect(td.get(0).props.children).toBe(1);
    expect(td.get(1).props.children).toBe('hermanmu@gmail.com');
    expect(td.get(2).props.children).toBe('michael');
    expect(td.get(3).props.children).toBe('true');
    expect(td.get(4).props.children).toBe('false');
});

test('UsersList renders a snapshot properly', () => {
    const tree = renderer.create(<UsersList users={users}/>).toJSON();
    expect(tree).toMatchSnapshot();
});