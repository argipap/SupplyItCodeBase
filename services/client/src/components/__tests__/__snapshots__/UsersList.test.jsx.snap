// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList renders a snapshot properly 1`] = `
<div>
  <h1
    className="title is-1"
  >
    All Users
  </h1>
  <hr />
  <br />
  <table
    className="table is-hoverable is-fullwidth"
  >
    <thead>
      <tr>
        <th>
          ID
        </th>
        <th>
          Email
        </th>
        <th>
          Username
        </th>
        <th>
          Active
        </th>
        <th>
          Admin
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          1
        </td>
        <td>
          hermanmu@gmail.com
        </td>
        <td>
          michael
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
      <tr>
        <td>
          2
        </td>
        <td>
          michael@mherman.org
        </td>
        <td>
          michaelherman
        </td>
        <td>
          true
        </td>
        <td>
          false
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
